Sources/MapboxMaps/Offline/OfflineErrors.swift

```diff
@@ -0,0 +1,110 @@
+import Foundation
+
+internal protocol CoreErrorRepresentable {
+    associatedtype CoreErrorType
+    init(coreError: CoreErrorType)
+    init(unspecifiedError: String)
+}
+
+/// Describes the reason for a tile region download request failure.
+public enum TileRegionError: LocalizedError, CoreErrorRepresentable {
+    typealias CoreErrorType = MapboxCommon.TileRegionError
+
+    /// The operation was canceled.
+    case canceled(String)
+
+    /// The tile region does not exist.
+    case doesNotExist(String)
+
+    /// Resolving the tileset descriptors failed.
+    case tilesetDescriptor(String)
+
+    /// There is no available space to store the resources.
+    case diskFull(String)
+
+    /// Some other failure reason.
+    case other(String)
+
+    internal init(coreError: MapboxCommon.TileRegionError) {
+        let message = coreError.message
+        switch coreError.type {
+        case .canceled:
+            self = .canceled(message)
+        case .doesNotExist:
+            self = .doesNotExist(message)
+        case .tilesetDescriptor:
+            self = .tilesetDescriptor(message)
+        case .diskFull:
+            self = .diskFull(message)
+        case .other:
+            self = .other(message)
+        }
+    }
+
+    public var errorDescription: String? {
+        switch self {
+        case let .canceled(message):
+            return ".canceled(\(message))"
+        case let .doesNotExist(message):
+            return ".doesNotExist(\(message))"
+        case let .tilesetDescriptor(message):
+            return ".tilesetDescriptor(\(message))"
+        case let .diskFull(message):
+            return ".diskFull(\(message))"
+        case let .other(message):
+            return ".other(\(message))"
+        }
+    }
+
+    internal init(unspecifiedError: String) {
+        self = .other(unspecifiedError)
+    }
```

Comment:
I think we should remove this and have a separate error type for platform-level errors.

Reply:
same for StylePackError

Reply:
Yep, I like this.